projects/igniteui-angular/src/lib/tabs/tabs.component.spec.ts

```diff
@@ -397,6 +412,65 @@ describe('IgxTabs', () => {
         const indicator = dom.query(By.css('.igx-tabs__header-menu-item-indicator'));
         expect(indicator.nativeElement.style.width).toBe('90px');
     }));
+
+    it('should navigate to the correct URL when clicking on tab buttons', fakeAsync(() => {
+        const router = TestBed.get(Router);
+        const location = TestBed.get(Location);
+        const fixture = TestBed.createComponent(TabsRoutingTestComponent);
+        const tabsComp = fixture.componentInstance.tabs;
+        fixture.detectChanges();
+
+        fixture.ngZone.run(() => { router.initialNavigation(); });
+        tick();
+        expect(location.path()).toBe('/');
+
+        const theTabs = tabsComp.contentTabs.toArray();
+
+        fixture.ngZone.run(() => { theTabs[2].nativeTabItem.nativeElement.dispatchEvent(new Event('click')); });
+        tick();
+        expect(location.path()).toBe('/view3');
+
+        fixture.ngZone.run(() => { theTabs[1].nativeTabItem.nativeElement.dispatchEvent(new Event('click')); });
+        tick();
+        expect(location.path()).toBe('/view2');
+
+        fixture.ngZone.run(() => { theTabs[0].nativeTabItem.nativeElement.dispatchEvent(new Event('click')); });
+        tick();
+        expect(location.path()).toBe('/view1');
+    }));
+
+    it('should select the correct tab button/panel when navigating an URL', fakeAsync(() => {
+        const router = TestBed.get(Router);
```

Comment:
Start using nested describes with a beforeEach creating the test component.

Reply:
Addressed.